espnet/nets/pytorch_backend/transformer/decoder.py

```diff
@@ -115,30 +115,42 @@ def forward(self, tgt, tgt_mask, memory, memory_mask):
             x = self.output_layer(x)
         return x, tgt_mask
 
-    def recognize(self, tgt, tgt_mask, memory):
+    def recognize(self, tgt, tgt_mask, memory, cache=None):
```

Comment:
Not directly related to this PR, but why don’t you change `recognize` to `forward_one_step`?

Reply:
I agree.